### PR TITLE
Anti-rug mechanism for JPEGs

### PR DESCRIPTION
**Pull request by vikiival**
:ambulance: half-fix for the random double spend

- :bug: (GalleryItem): removed unsued imports
- :zap: added rmrk grapghql client
- :zap: added Inmemorry cache
- :zap: refactored transaction executor into separated functions
- :zap: added checker before buy

Thank you for your contribution to the KodaDot NFT gallery, 
we really appreciate your contribution!

### PR type
- [x] Bugfix
- [x] Feature
- [x] Refactoring

### Before submitting this PR, please make sure:
- [x] Your code builds clean without any erros or warnigns
- [ ] You've posted screenshot of demonstrated change in this PR
- [x] Merged recent default branch, **main** and you have no conflicts
- [ ] Didn't break any original functionality 

### Optional
- [ ] You've tested it on mobile
- [ ] Are there any edge cases? Name if any

### CODE REVIEW

File: src/components/rmrk/Gallery/AvailableActions.vue [hunk 1]
```diff
@@ -176,6 +180,33 @@ export default class AvailableActions extends Mixins(RmrkVersionMixin) {
...
 
+  protected async isTransactionPending() {
+    try {
+      const nft = await this.$apollo.query<{ nfts: PendingTransaction[] }>({
+        client: 'rmrkApolloClient',
+        query: verificationGuard,
+        variables: {
+          id: this.nftId
+        },
+      })
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/AvailableActions.vue:191]: What does this do? first time seing this weird syntax

File: src/components/rmrk/Gallery/AvailableActions.vue [hunk 2]
```diff
@@ -176,6 +180,33 @@ export default class AvailableActions extends Mixins(RmrkVersionMixin) {
 
   }
 
+  protected async isTransactionPending() {
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/AvailableActions.vue:183]: return type?

File: src/components/rmrk/Gallery/AvailableActions.vue [hunk 3]
```diff
@@ -176,6 +180,33 @@ export default class AvailableActions extends Mixins(RmrkVersionMixin) {
...
+      const nft = await this.$apollo.query<{ nfts: PendingTransaction[] }>({
+        client: 'rmrkApolloClient',
+        query: verificationGuard,
+        variables: {
+          id: this.nftId
+        },
+      })
+
+      const { data: {nfts} } = nft
+      console.log('nfts', nfts, nft)
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/AvailableActions.vue:194]: console log here

**Comment by yangwao** [comment]
👀 what if we can release it today

**Review by roiLeo — APPROVED** [review]

File: src/components/rmrk/Gallery/AvailableActions.vue (hunk 1 above)
Comment (vikiival) [reply src/components/rmrk/Gallery/AvailableActions.vue:191]: > weird syntax

by default is `query` function of type `T` and you can say to it what's the expected return type

**Pull request merged**